kindr_ros/CMakeLists.txt

```diff
@@ -1,57 +1,68 @@
-cmake_minimum_required(VERSION 3.5.1)
+cmake_minimum_required(VERSION 3.5)
 project(kindr_ros)
 
-add_definitions(-std=c++11)
-
-find_package(catkin REQUIRED COMPONENTS
-  geometry_msgs tf
-  )
-
-# Attempt to find catkinized kindr
-find_package(kindr QUIET)
-if(NOT kindr_FOUND)
-  # Attempt to find package-based kindr
-  find_package(PkgConfig REQUIRED)
-  pkg_check_modules(kindr kindr REQUIRED)
+# Use C++14 standard
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
 endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(kindr REQUIRED)
 
+find_package(eigen3_cmake_module REQUIRED)
 find_package(Eigen3 REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS include ${EIGEN3_INCLUDE_DIR}
-  #  LIBRARIES kindr_ros
-  CATKIN_DEPENDS geometry_msgs tf
-  DEPENDS kindr
+set(dependencies
+  eigen3_cmake_module
+  Eigen3
+  tf2
+  geometry_msgs
 )
 
-include_directories(include)
-include_directories(${catkin_INCLUDE_DIRS})
-include_directories(${EIGEN3_INCLUDE_DIR})
-include_directories(${kindr_INCLUDE_DIRS})
+install(DIRECTORY include/${PROJECT_NAME}/
+        DESTINATION include/${PROJECT_NAME})
+
+if(BUILD_TESTING)
+  find_package(ament_lint_auto REQUIRED)
 
-#############
-## Testing ##
-#############
+  ament_lint_auto_find_test_dependencies()
 
-if(CATKIN_ENABLE_TESTING)
-  catkin_add_gtest(${PROJECT_NAME}-test
+  find_package(ament_cmake_gtest REQUIRED)
+
+  ament_add_gtest(${PROJECT_NAME}-test
     test/test_main.cpp
     test/RosGeometryMsgPhysicalQuantitiesTest.cpp
     test/RosGeometryMsgRotationTest.cpp
     test/RosGeometryMsgPoseTest.cpp
     test/RosTfPoseTest.cpp
     test/TfConventionTest.cpp
     WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test
-    )
+  )
+
   if(TARGET ${PROJECT_NAME}-test)
-    target_link_libraries(${PROJECT_NAME}-test ${catkin_LIBRARIES})
-    find_package(cmake_code_coverage QUIET)
-    if(cmake_code_coverage_FOUND)
-      add_gtest_coverage(TEST_BUILD_TARGETS ${PROJECT_NAME}-test)
-    endif(cmake_code_coverage_FOUND)
+    target_include_directories(${PROJECT_NAME}-test PUBLIC
+      include
+      ${EIGEN3_INCLUDE_DIRS})
+    ament_target_dependencies(${PROJECT_NAME}-test
+      ${dependencies}
+    )
+    # TODO(SivertHavso): add back code coverage:
+    #   find_package(cmake_code_coverage QUIET)
+    #   if(cmake_code_coverage_FOUND)
+    #     add_gtest_coverage(TEST_BUILD_TARGETS ${PROJECT_NAME}-test)
+    #   endif()
```

Comment:
It will have to wait until I've finished porting the elevation_mapping package. I'd say merge it now since it "only" affects quality assurance and not the API or ABI. The tests themself still run and there shouldn't be a notable difference in code coverage between the ROS1 and ROS2 branch for now.

Reply:
Out of curiosity, why is this a dependency of the porting of `elevation_mapping`? This is a downstream dependency, right?